include/limits.h

```diff
@@ -146,8 +146,8 @@
 
 /* Required for sigqueue */
 
-#define _POSIX_RTSIG_MAX      2    /*  Number of reserved realtime signals */
-#define _POSIX_SIGQUEUE_MAX   32
+#define _POSIX_RTSIG_MAX      32   /*  Number of reserved realtime signals */
```

Comment:
https://pubs.opengroup.org/onlinepubs/7908799/xsh/limits.h.html defines this to 8 and `RTSIG_MAX` to be a minimum of `_POSIX_RTSIG_MAX`.

I Linux case things are a bit strange as:
1. `_POSIX_RTSIG_MAX` is 8 (that is actually only one non-strange thing)
2. `RTSIG_MAX` is 32.
3. `SIGRTMAX - SIGRTMIN` is 30, but I was expecting it to be equal to `RTSIG_MAX` actually

Reply:
Opengroup.org is not consistent.  Yes I saw that page that uses eight.  Other pages specify 16 and 64.  The most recent ones try to show a range of values {min/max}.  Linux defines this as 32.  I conclude that OpenGroup has no specification for any exact number of real time signals.

 I am following Linux.  I think that is the right thing to do.  See https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/limits.h#L19

I mention this in #8869

Reply:
Personally I think that `_POSIX_RTSIG_MAX` can be a minimum number of realtime signals, but we can have more.
We already have similar logic in
```
#if CONFIG_LIBC_OPEN_MAX < _POSIX_OPEN_MAX
#  define OPEN_MAX     _POSIX_OPEN_MAX
#else
#  define OPEN_MAX     CONFIG_LIBC_OPEN_MAX
#endif
```
So the same with `RTSIG_MAX`. It can be a minimum of `_POSIX_RTSIG_MAX` and other value (32). As it is not worth of config option really, so we can define `RTSIG_MAX` to 32.

Reply:
The same for `_POSIX_SIGQUEUE_MAX` and `SIGQUEUE_MAX`

Reply:
BTW, Linux even do not define `SIGQUEUE_MAX`

Reply:
> As it is not worth of config option really, so we can define `RTSIG_MAX` to 32.

32 is a LOT of real time signals!  But that is what Linux uses 32 and since sigset_t is based on uint32_t's, nothing would be saved by arbitrarily limiting the user to fewer signals.  So I think it should be 32, but I don't have a super strong feeling about that.  I will resolve this now... Thanks for bringing this up.

Reply:
As I already stated in one of the above comments, the Linux case is inconsistent (30 RT signals based on MAX and MIN diff while 32 are declared)

Reply:
I would recommend to "blindly follow" POSIX here and define the base POSIX variant to 8, but not use it as the base for system internals